nifi-nar-bundles/nifi-gcp-bundle/nifi-gcp-processors/src/main/java/org/apache/nifi/processors/gcp/bigquery/PutBigQueryStreaming.java

```diff
@@ -191,6 +192,18 @@ public void onTrigger(ProcessContext context, ProcessSession session) throws Pro
                     lmapr.add(convertMapRecord(((MapRecord) mapr).toMap()));
                 }
                 result.put(key, lmapr);
+            } else if (obj instanceof Timestamp) {
+                SimpleDateFormat format = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+                format.setTimeZone(TimeZone.getTimeZone("UTC"));
+                result.put(key, format.format(((Timestamp) obj).getTime()));
+            } else if (obj instanceof Time) {
+                SimpleDateFormat format = new SimpleDateFormat("HH:mm:ss");
+                format.setTimeZone(TimeZone.getTimeZone("UTC"));
+                result.put(key, format.format(((Time) obj).getTime()));
+            } else if (obj instanceof Date) {
+                SimpleDateFormat format = new SimpleDateFormat("yyyy-MM-dd");
+                format.setTimeZone(TimeZone.getTimeZone("UTC"));
+                result.put(key, format.format(((Date) obj).getTime()));
```

Comment:
When executing the integration test you added, I see the below data being added in BQ:

<img width="858" alt="Screenshot 2021-07-17 at 02 01 35" src="https://user-images.githubusercontent.com/11541012/126018935-4da2b128-6c1a-4538-a5a2-4dd1a8e6a490.png">

Which makes the test fail because:
````java
assertEquals(john.get("birth").getRecordValue().get(0).getStringValue(), "1995-06-01");
assertEquals(john.get("birth").getRecordValue().get(1).getStringValue(), "15:57:27");
assertEquals(john.get("birth").getRecordValue().get(2).getTimestampValue() / 1000, 802022247000L);
````

Something needs to be fixed.